sys/include/stdio_base.h

```diff
@@ -116,18 +119,59 @@ ssize_t stdio_read(void* buffer, size_t max_len);
 /**
  * @brief write @p len bytes from @p buffer into STDOUT
  *
- * @note Depending on the stdio backend(s) used, not all bytes might
- * be written to stdout and accounted for if multiple backends are
- * active, as not all stdout backends will do a blocking write.
- *
  * @param[in]   buffer  buffer to read from
- * @param[in]   len     nr of bytes to write
+ * @param[in]   len     number of bytes to write
  *
- * @return nr of bytes written
+ * @return number of bytes written
  * @return <0 on error
+ *
+ * If not all requested bytes were written, the caller should retry with the
+ * remaining bytes in a busy loop. At latest on the second call, the
+ * implementation will likely block.
```

Comment:
```suggestion
 * implementation will likely block, but should not block indefinitely.
```

Reply:
To some extent, it probably depends on the implementation when the stdio should fail: A stdio that depends on another thread to run will fail when running in an interrupts or when interrupts are off, but may block otherwise. A stdio that depends on Rust async stuff may need to do more checks, and must not block if called from an async context that prevents its own worker from running.

I don't think we can tell more here; the stdio implementation will need to know from its own dependencies what may cause a deadlock.